drivers/net/ppp.c

```diff
@@ -698,10 +752,97 @@ static struct net_stats_ppp *ppp_get_stats(struct device *dev)
 }
 #endif
 
+#if !defined(CONFIG_NET_TEST)
+static void ppp_uart_flush(struct device *dev)
+{
+	u8_t c;
+
+	while (uart_fifo_read(dev, &c, 1) > 0) {
+		continue;
+	}
+}
+
+static void ppp_uart_isr(void *user_data)
+{
+	struct ppp_driver_context *context = user_data;
+	struct device *uart = context->dev;
+	int rx = 0, ret;
+
+	/* get all of the data off UART as fast as we can */
+	while (uart_irq_update(uart) && uart_irq_rx_ready(uart)) {
+		rx = uart_fifo_read(uart, context->buf, sizeof(context->buf));
+		if (rx <= 0) {
+			continue;
+		}
+
+		ret = ring_buf_put(&context->rx_ringbuf, context->buf, rx);
+		if (ret < rx) {
+			LOG_ERR("Rx buffer doesn't have enough space. "
+				"Bytes pending: %d, written: %d",
+				rx, ret);
+			break;
```

Comment:
If the ring buffer does not have enough space, this seems to mean that those bytes are lost forever, since the next time this function is called new bytes are read from the fifo and the previous ones in the buffer overwritten. Do we loose sync and get stuck in ppp since the next fifo read will be glued to the end of the previous bytes for which the ring buffer had space?

Reply:
We might loose sync temporarily in this case but ppp will not get stuck as next received sync byte 0x7e will start the frame again.